flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -312,7 +313,7 @@ public void validateRunsInMainThread() {
 	/**
 	 * Executor which executes runnables in the main thread context.
 	 */
-	protected static class MainThreadExecutor implements Executor {
+	protected static class MainThreadExecutor implements MainThreadExecutable {
```

Comment:
`MainThreadExecutable` was actually intended to be a marker interface for the `RpcServer`. I would recommend to let the `MainThreadExecutor` implement the `ScheduledExecutor`. That way, there is no need to change the `MainThreadExecutable`.

Reply:
I was also thinking about that. Are you suggesting to replace the current methods by the ones from `ScheduledExecutor` or to add them on top of the existing methods?